File: GNSDK/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/GnTitle.java

package com.gracenote.gnsdk;

/** 
* Title of a work or product 
*/ 
 
public class GnTitle extends GnDataObject {
  private long swigCPtr;

  protected GnTitle(long cPtr, boolean cMemoryOwn) {
    super(gnsdk_javaJNI.GnTitle_SWIGUpcast(cPtr), cMemoryOwn);
    swigCPtr = cPtr;
  }

  protected static long getCPtr(GnTitle obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_GnTitle(swigCPtr);
      }
      swigCPtr = 0;
    }
    super.delete();
  }

  public static String gnType() {
    return gnsdk_javaJNI.GnTitle_gnType();
  }

  public static GnTitle from(GnDataObject obj) throws com.gracenote.gnsdk.GnException {
    return new GnTitle(gnsdk_javaJNI.GnTitle_from(GnDataObject.getCPtr(obj), obj), true);
  }

/** 
*  Title display language 
*  @return Language string 
*/ 
 
  public String language() {
    return gnsdk_javaJNI.GnTitle_language(swigCPtr, this);
  }

/** 
*  3 letter ISO code for display language 
*  @return Language code 
*/ 
 
  public String languageCode() {
    return gnsdk_javaJNI.GnTitle_languageCode(swigCPtr, this);
  }

/** 
*  Title display string 
*  @return Strng suitable for displaying to end user 
*/ 
 
  public String display() {
    return gnsdk_javaJNI.GnTitle_display(swigCPtr, this);
  }

/** 
*  Title prefix, e.g., The 
*  @return Prefix 
*/ 
 
  public String prefix() {
    return gnsdk_javaJNI.GnTitle_prefix(swigCPtr, this);
  }

/** 
*  Sortable title 
*  @return Sortable string 
*/ 
 
  public String sortable() {
    return gnsdk_javaJNI.GnTitle_sortable(swigCPtr, this);
  }

/** 
*  Sortable title scheme 
*  @return Sortable scheme 
*/ 
 
  public String sortableScheme() {
    return gnsdk_javaJNI.GnTitle_sortableScheme(swigCPtr, this);
  }

/** 
* Main title 
* @return Title 
*/ 
 
  public String mainTitle() {
    return gnsdk_javaJNI.GnTitle_mainTitle(swigCPtr, this);
  }

/** 
* Title edition 
* @return Edition 
*/ 
 
  public String edition() {
    return gnsdk_javaJNI.GnTitle_edition(swigCPtr, this);
  }

}

File: GNSDK/wrappers/gnsdk_csharp/src_wrapper/GnTitle.cs

namespace GracenoteSDK {

using System;
using System.Runtime.InteropServices;

/**
* Title of a work or product
*/
public class GnTitle : GnDataObject {
  private HandleRef swigCPtr;

  internal GnTitle(IntPtr cPtr, bool cMemoryOwn) : base(gnsdk_csharp_marshalPINVOKE.GnTitle_SWIGUpcast(cPtr), cMemoryOwn) {
    swigCPtr = new HandleRef(this, cPtr);
  }

  internal static HandleRef getCPtr(GnTitle obj) {
    return (obj == null) ? new HandleRef(null, IntPtr.Zero) : obj.swigCPtr;
  }

  ~GnTitle() {
    Dispose();
  }

  public override void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_GnTitle(swigCPtr);
        }
        swigCPtr = new HandleRef(null, IntPtr.Zero);
      }
      GC.SuppressFinalize(this);
      base.Dispose();
    }
  }

  public static string GnType() {
    string ret = gnsdk_csharp_marshalPINVOKE.GnTitle_GnType();
    return ret;
  }

  public static GnTitle From(GnDataObject obj) {
    GnTitle ret = new GnTitle(gnsdk_csharp_marshalPINVOKE.GnTitle_From(GnDataObject.getCPtr(obj)), true);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
    return ret;
  }

/**
*  Title display string
*  @return Strng suitable for displaying to end user
*/
  public string Display {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnTitle_Display_get(swigCPtr) );
	} 

  }

/**
*  Title display language
*  @return Language string
*/
  public string Language {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnTitle_Language_get(swigCPtr) );
	} 

  }

/**
*  3 letter ISO code for display language
*  @return Language code
*/
  public string LanguageCode {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnTitle_LanguageCode_get(swigCPtr) );
	} 

  }

/**
*  Sortable title
*  @return Sortable string
*/
  public string Sortable {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnTitle_Sortable_get(swigCPtr) );
	} 

  }

/**
*  Sortable title scheme
*  @return Sortable scheme
*/
  public string SortableScheme {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnTitle_SortableScheme_get(swigCPtr) );
	} 

  }

/**
* Title edition
* @return Edition
*/
  public string Edition {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnTitle_Edition_get(swigCPtr) );
	} 

  }

/**
* Main title
* @return Title
*/
  public string MainTitle {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnTitle_MainTitle_get(swigCPtr) );
	} 

  }

/**
*  Title prefix, e.g., The
*  @return Prefix
*/
  public string Prefix {
	get
	{   
		/* csvarout typemap code */
		return GnMarshalUTF8.StringFromNativeUtf8(gnsdk_csharp_marshalPINVOKE.GnTitle_Prefix_get(swigCPtr) );
	} 

  }

}

}
